action.yml

```diff
@@ -20,5 +20,46 @@ inputs:
     default: 'false'
 
 runs:
-  using: 'node20'
-  main: 'dist/invoke-action.mjs'
+  using: "composite"
+  steps:
+    - id: version
+      shell: bash
+      run: echo "version=v0.2.1" >> "${GITHUB_OUTPUT}"
+    - id: arch
+      shell: bash
+      run: |
+        if [ ${{ runner.arch }} = "X64" ]; then
+          echo "arch=x86_64" >> "${GITHUB_OUTPUT}"
+        elif [ ${{ runner.arch }} = "X86" ]; then
+          echo "arch=i386" >> "${GITHUB_OUTPUT}"
+        else
+          echo "arch=${{ runner.arch }}" >> "${GITHUB_OUTPUT}"
+        fi
+    - id: ext
+      shell: bash
+      run: |
+        if [ ${{ runner.os }} = "Windows" ]; then
+          echo "ext=zip" >> "${GITHUB_OUTPUT}"
+        else
+          echo "ext=tar.gz" >> "${GITHUB_OUTPUT}"
+        fi
+    - name: fetch in Linux
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        wget "https://github.com/AtomicFS/firmware-action/releases/download/${{ steps.version.outputs.version }}/firmware-action_${{ runner.os }}_${{ steps.arch.outputs.arch }}.${{ steps.ext.outputs.ext }}"
+        tar -xvzf "firmware-action_${{ runner.os }}_${{ steps.arch.outputs.arch }}.${{ steps.ext.outputs.ext }}"
+    - name: fetch in Windows
+      if: runner.os == 'Windows'
+      shell: bash
+      run: |
+        echo "use Linux"
```

Comment:
Does wget and tar work the same in macOS as you're only checking for Windows? Generally it would be good to keep Windows Runner compatibility and as only tar and something to fetch is needed we could probably substitute it here with something more Windows-like.

Reply:
I fixed it now so that the action should work on all systems currently available - meaning it can download and run the executable on `Linux`, `Windows` and `MacOS`.

Running the `firmware-action` executable fails on `Windows` because of JSON validation, and fails on `MaOS` because Docker is not properly installed. On windows it is just matter of creting a new windows-specific config with windows paths probably, so that the JSON validator is satisfied. As for MacOS, I have tried to install Docker into the GitHub virtual machine, but with no success.

As it is right now, I am not going to fix these issues since we do not have any real use-case for them. However with these new fixes the functionality of JavaScript-based action should be preserved.